beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockFactoryTest.java

```diff
@@ -404,14 +418,26 @@ private BlockFactory createBlockFactory(final Spec spec) {
   }
 
   private void prepareDefaultPayload(final Spec spec) {
-    executionPayload =
-        SchemaDefinitionsBellatrix.required(spec.getGenesisSpec().getSchemaDefinitions())
-            .getExecutionPayloadSchema()
-            .getDefault();
-
-    executionPayloadHeader =
-        SchemaDefinitionsBellatrix.required(spec.getGenesisSpec().getSchemaDefinitions())
-            .getExecutionPayloadHeaderSchema()
-            .getHeaderOfDefaultPayload();
+    if (spec.isMilestoneSupported(SpecMilestone.CAPELLA)) {
```

Comment:
Shouldn't need this bit at all because `SchemaDefinitionsBellatrix.required` should work even if the genesis spec is Capella (`SchemaDefinitionsCapella` should implement `SchemaDefinitionsBellatrix`) and it would just return the Capella execution payload schemas. If that's not working, we should fix the capella schema definitions impl.

Reply:
ah I had partially removed some changes like that but not this one. fixed.